### ISSUE TITLE
Keyboard activation of drag and drop in react-aria-components Table fails with "can't access property key, column is undefined"

### ISSUE THREAD

**Issue by kwang1083** [adobe/react-spectrum#9000]
### Provide a general summary of the issue here

For the react-aria-components `Table` with the `useDragAndDrop` hook, when using keyboard navigation to select the drag and drop indicator, the selection fails with the error: "can't access property key, column is undefined". However, using the mouse to drag and drop works properly. The root cause appears to be that the `cell` variable at [L193](https://github.com/adobe/react-spectrum/blob/ee53a795f4a2200b8bac9d3cb2082e5ee62ce6d6/packages/react-aria-components/src/Table.tsx#L193) incorrectly contains a `row` during keyboard selection, which means the index of the row could be greater than the max size of `columns`. This results in an undefined column and the error above.

In correct calls of getTextValue, cell is set to type = "cell" and the index is bounded by column size: 

<img width="1407" height="431" alt="Image" src="https://github.com/user-attachments/assets/c6c2e4a9-b8b8-4b47-9598-6cc1926230b5" />

On keyboard selection, cell is set to type = "item", representing a row of data, and the index is bounded incorrectly by row size:

<img width="1407" height="477" alt="Image" src="https://github.com/user-attachments/assets/2c7b04a0-8dfd-4d2e-b7b4-41b795b9e8c7" />

Previously, I used react-aria-components v1.4.1 which didn't have this issue. When upgrading to react-aria-components 1.12.1, I ran into this issue. 

### 🤔 Expected Behavior?

The expected behavior is what is shown on the documentation site at: https://react-spectrum.adobe.com/react-aria/Table.html#reorderable. Keyboard navigation and selection of the drag and drop indicator should work properly with "enter" for selection and arrow keys to drag and drop.

### 😯 Current Behavior

Currently, keyboard selection throws an error for drag and drop when the number of rows is greater than the number of columns. This is because the index of the "row" is used instead of the index of the "cell" on keyboard selection, and columns[row.index] can return undefined if the number of rows is greater than the number of columns

### 💁 Possible Solution

A quick solution be a check for cell.type to make sure it is "cell" rather than "item", which can be added  to [Table.tsx code](https://github.com/adobe/react-spectrum/blob/ee53a795f4a2200b8bac9d3cb2082e5ee62ce6d6/packages/react-aria-components/src/Table.tsx#L195). Another check could be ensuring the row computed in getTextValue has type "item" instead of "tablebody". 

However, this wouldn't address the root problem of why the tablebody + row is being accessed instead of the row + cell inside the function, on keyboard selection only (mouse interactions work correctly). I'm hoping the team would have more insight on what causes this behavior and potential fixes!

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/devbox/nostalgic-snyder-k3qcy7?workspaceId=ws_TPcixFMV8nsJkNho7e7pFT

This code sandbox shows the example from https://react-spectrum.adobe.com/react-aria/Table.html#reorderable with two additional rows added. When using keyboard to navigate to the drag + drop indicator and selecting with "enter", you will see the error. Using latest version of react-aria-components.

### Version

V1.13 of react-aria-components

### What browsers are you seeing the problem on?

Firefox

### If other, please specify.

Chrome as well, likely others

### What operating system are you using?

MacOS

**Comment by snowystinger** [comment]
This has a workaround, which is to provide `textValue` to the row.

I'm not sure why it's trying to get the textValue for body. Will need to chase it farther up the trace to see why that's happening. I think it's find to leave the `getTextValue` alone and we shouldn't call it on the body key.
From a quick look, it would appear that useDropIndicator is asking for prevKey of a row, which is returning the body. I do not think that it should be doing that. So we may need to query something other than the collection so that the returned prevKey is null. (potentially same with nextKey)

@LFDanLu can you double check me? I assume we should have access to the keyboard or layout delegate for this somewhere, but I'm not seeing it

**Comment by chirokas** [comment]
related to #8645

**Comment by snowystinger** [comment]
o, thank you @chirokas ! I have added that PR to our backlog to try to get some priority for it.

**Comment by kwang1083** [comment]
Thank you for taking a quick look @snowystinger! I'll try and integrate the workaround to avoid this bug for now. Also, thanks for the work on your pr @chirokas. Skipping non node items like tablebody should fix this error here, excited for the pr to get merged!
